Modules/CLI/CreateDICOMSeries/CreateDICOMSeries.cxx

```diff
@@ -222,6 +226,49 @@ int DoIt( int argc, char * argv[])
       {
       itk::EncapsulateMetaData<std::string>(dictionary, "0008|103e", seriesDescription);
       }
+    if( seriesDate.size() > 0 )
+      {
+      itk::EncapsulateMetaData<std::string>(dictionary, "0008|0021", seriesDate);
+      }
+    if( seriesTime.size() > 0 )
+      {
+      itk::EncapsulateMetaData<std::string>(dictionary, "0008|0031", seriesTime);
+      }
+    if( contentDate.size() > 0 )
+      {
+      itk::EncapsulateMetaData<std::string>(dictionary, "0008|0023", contentDate);
+      }
+    if( contentTime.size() > 0 )
+      {
+      itk::EncapsulateMetaData<std::string>(dictionary, "0008|0033", contentTime);
+      }
+
+    bool keepOriginalUID = (!studyInstanceUID.empty() || !seriesInstanceUID.empty() || !frameOfReferenceInstanceUID.empty());
+    gdcmIO->SetKeepOriginalUID(keepOriginalUID);
+    if( studyInstanceUID.size() > 0 )
+      {
+      itk::EncapsulateMetaData<std::string>(dictionary, "0020|000d", studyInstanceUID);
+      }
+    else if (keepOriginalUID)
+      {
+      std::cerr << "studyInstanceUID must be specified if any of the UIDs are specified." << std::endl;
```

Comment:
Should there be a `return` here? If this is a warning, then maybe the message should not say "must"? Also, I do not understand the logic why SeriesInstanceUID would be required if StudyInstanceUID or FrameOfReferenceUID are specified? Can you comment?

Reply:
If you prefer, we can return with error.

Reply:
It is just confusing that the message is not consistent with the action. I would either return error, but probably better to just print a warning, because I don't see a good reason why this should be a "must".

Reply:
> I don't see a good reason why this should be a "must"

[itkGDCMImageIO](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Modules/IO/GDCM/src/itkGDCMImageIO.cxx#L1154-L1189) implements an all-or-nothing approach with these 3 UIDs. You either specify them or ITK generates all of them.

Reply:
I see. Thanks for investigating. I don't understand that logic, but it is unrelated to this PR.